monai/utils/radial_nms.py

```diff
@@ -0,0 +1,55 @@
+from typing import Union
+
+import numpy as np
+import torch
+
+from monai.networks.layers import GaussianFilter
+
+
+class RadialNMS:
+    def __init__(self, spatial_dims: int, sigma: float = 0.0, prob_threshold: float = 0.5, radius: int = 24):
+        self.sigma = sigma
+        self.spatial_dims = spatial_dims
+        if self.sigma > 0:
+            self.filter = GaussianFilter(spatial_dims=spatial_dims, sigma=sigma)
+        self.prob_threshold = prob_threshold
+        self.radius = radius
+
+    def __call__(self, probs_map: Union[np.ndarray, torch.Tensor], level: int):
+        if self.sigma > 0:
+            if isinstance(probs_map, np.ndarray):
+                probs_map = torch.as_tensor(probs_map, dtype=torch.float)
+            device = (
+                torch.device("cuda")
+                if (probs_map.device == "cuda" and torch.cuda.is_available())
+                else torch.device("cpu:0")
+            )
+            self.filter.to(device)
```

Comment:
I think this could be simplified to 
```suggestion
            self.filter.to(probs_map)
```

Reply:
Exactly, this way it uses whatever device that `prob_maps` is residing on.

Reply:
Thanks! I took https://github.com/Project-MONAI/MONAI/blob/master/tests/test_gaussian_filter.py#L41 for reference. Do I need to modify that place too?

Reply:
I don't think they are the same, that one is to select based on a string, here is based on an existing variable...